src/System.IO.Pipelines/ref/System.IO.Pipelines.cs

```diff
@@ -33,25 +27,16 @@ public sealed partial class Pipe
         public System.IO.Pipelines.PipeWriter Writer { get { throw null; } }
         public void Reset() { }
     }
-    public partial struct PipeAwaiter<T> : System.Runtime.CompilerServices.ICriticalNotifyCompletion, System.Runtime.CompilerServices.INotifyCompletion
-    {
-        private object _dummy;
-        public PipeAwaiter(System.IO.Pipelines.IPipeAwaiter<T> awaiter) { throw null; }
-        public bool IsCompleted { get { throw null; } }
-        public System.IO.Pipelines.PipeAwaiter<T> GetAwaiter() { throw null; }
-        public T GetResult() { throw null; }
-        public void OnCompleted(System.Action continuation) { }
-        public void UnsafeOnCompleted(System.Action continuation) { }
-    }
     public partial class PipeOptions
     {
-        public PipeOptions(System.Buffers.MemoryPool<byte> pool = null, System.IO.Pipelines.PipeScheduler readerScheduler = null, System.IO.Pipelines.PipeScheduler writerScheduler = null, long pauseWriterThreshold = (long)0, long resumeWriterThreshold = (long)0, int minimumSegmentSize = 2048) { }
+        public PipeOptions(System.Buffers.MemoryPool<byte> pool = null, System.IO.Pipelines.PipeScheduler readerScheduler = null, System.IO.Pipelines.PipeScheduler writerScheduler = null, long pauseWriterThreshold = (long)0, long resumeWriterThreshold = (long)0, int minimumSegmentSize = 2048, bool useSynchronizationContext = true) { }
         public static System.IO.Pipelines.PipeOptions Default { get { throw null; } }
         public int MinimumSegmentSize { get { throw null; } }
         public long PauseWriterThreshold { get { throw null; } }
         public System.Buffers.MemoryPool<byte> Pool { get { throw null; } }
         public System.IO.Pipelines.PipeScheduler ReaderScheduler { get { throw null; } }
         public long ResumeWriterThreshold { get { throw null; } }
+        public bool UseSynchronizationContext { get { throw null; } }
```

Comment:
This is a new API I added to reason about capturing the synchronization context vs preferring the specified scheduler if both exist. By default this is true so it does the right thing. Turning it off is trivial as well.